crates/router/src/connector/paybox/transformers.rs

```diff
@@ -546,38 +644,68 @@ impl<F>
+                            })),
+                            network_txn_id: None,
+                            connector_response_reference_id: None,
+                            incremental_authorization_allowed: None,
+                            charge_id: None,
+                        }),
+                        ..item.data
+                    }),
+                    false => Ok(Self {
+                        response: Err(types::ErrorResponse {
+                            code: response.response_code.clone(),
+                            message: response.response_message.clone(),
+                            reason: Some(response.response_message),
+                            status_code: item.http_code,
+                            attempt_status: None,
+                            connector_transaction_id: Some(response.transaction_number),
+                        }),
+                        ..item.data
+                    }),
+                }
+            }
+            PayboxResponse::ThreeDs(data) => Ok(Self {
+                status: enums::AttemptStatus::AuthenticationPending,
                 response: Ok(types::PaymentsResponseData::TransactionResponse {
-                    resource_id: types::ResponseId::ConnectorTransactionId(
-                        response.paybox_order_id,
-                    ),
-                    redirection_data: None,
+                    resource_id: types::ResponseId::NoResponseId,
+                    redirection_data: Some(RedirectForm::Html {
+                        html_data: data.peek().to_string(),
+                    }),
                     mandate_reference: None,
-                    connector_metadata: Some(serde_json::json!(PayboxMeta {
-                        connector_request_id: response.transaction_number.clone()
-                    })),
+                    connector_metadata: None,
                     network_txn_id: None,
                     connector_response_reference_id: None,
                     incremental_authorization_allowed: None,
                     charge_id: None,
                 }),
                 ..item.data
             }),
-            false => Ok(Self {
+            PayboxResponse::Error(_) => Ok(Self {
                 response: Err(types::ErrorResponse {
-                    code: response.response_code.clone(),
-                    message: response.response_message.clone(),
-                    reason: Some(response.response_message),
+                    code: consts::NO_ERROR_CODE.to_string(),
+                    message: consts::NO_ERROR_MESSAGE.to_string(),
+                    reason: Some(consts::NO_ERROR_MESSAGE.to_string()),
                     status_code: item.http_code,
                     attempt_status: None,
-                    connector_transaction_id: Some(item.response.transaction_number),
+                    connector_transaction_id: None,
```

Comment:
Why aren't we populating the error values here?

Reply:
This will only come when we are getting Error in html response